lr-cli/cortx_setup/commands/storage/config.py

```diff
@@ -0,0 +1,251 @@
+#
+# Copyright (c) 2020 Seagate Technology LLC and/or its Affiliates
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as published
+# by the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU Affero General Public License for more details.
+# You should have received a copy of the GNU Affero General Public License
+# along with this program. If not, see <https://www.gnu.org/licenses/>.
+# For any questions about this software or licensing,
+# please email opensource@seagate.com or cortx-questions@seagate.com.
+#
+
+from pathlib import Path
+from ..command import Command
+from cortx.utils.conf_store import Conf
+from provisioner.commands import PillarSet
+from provisioner.salt import local_minion_id
+from .commons import Commons
+
+#TODO: Add this path in the global config
+prvsnr_cluster_path = Path(
+    '/opt/seagate/cortx_configs/provisioner_cluster.json'
+)
+
+
+"""Cortx Setup API for configuring the storage enclosure """
+
+
+class StorageEnclosureConfig(Command):
+    '''
+    $ cortx_setup storage config --name <enclosure-name>
+    $ cortx_setup storage config --type {RBOD|JBOD|EBOD|virtual}
+    $ cortx_setup storage config --user <user> --password <password>
+    $ cortx_setup storage config --controller_type {gallium|indium|virtual}
+    $ cortx_setup storage config --mode {primary|secondary} --ip <ip-address> --port <port-number>
+    '''
+    _args = {
+        'name': {
+            'type': str,
+            'default': None,
+            'optional': True,
+            'help': 'Set name of the enclosure e.g enc1-r10'
+        },
+        'type': {
```

Comment:
Set `dest` on this argument to avoid type in 
```def run(self, name=None, type=None,```
Check with @SomwanshiAnjali

Reply:
addressed it by using **kwargs